plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -242,6 +267,200 @@ export class GithubEntityProvider implements EntityProvider {
       presence: 'optional',
     };
   }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.onEvent} */
+  async onEvent(params: EventParams): Promise<void> {
+    this.logger.debug(`Received event from ${params.topic}`);
+    if (params.topic !== TOPIC_REPO_PUSH) {
+      return;
+    }
+
+    if (params.metadata?.['x-github-event'] === 'push') {
+      await this.onRepoPush(params.eventPayload as PushEvent);
+    }
+
+    return;
+  }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.supportsEventTopics} */
+  supportsEventTopics(): string[] {
+    return [TOPIC_REPO_PUSH];
+  }
+
+  private async onRepoPush(event: PushEvent) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const repoName = event.repository.name;
+    const repoUrl = event.repository.url;
+    this.logger.debug(`handle github:push event for ${repoName} - ${repoUrl}`);
+
+    const branch =
+      this.config.filters?.branch || event.repository.default_branch;
+
+    if (!event.ref.includes(branch)) {
+      this.logger.debug(`skipping push event from ref ${event.ref}`);
+      return;
+    }
+
+    if (this.config.checkRepositoryFiltersForWebhook) {
+      const repository = await this.getRepositoryInfo(repoName);
```

Comment:
Is event.repository not containing all required information? I saw that `checkRepositoryFiltersForWebhook` defaults to `false`. However, I think it is rather expected to be true. Otherwise, you may receive events from repositories which you actually wanted to ignore based on your filter config.

I get that this is an optimization to omit the additional API call. I'd leave this rather as an opt-in though.

Reply:
🤔 I will check again, maybe I lost something here

Reply:
You are both right, I updated to avoid the request